src/nix-eval-jobs.cc

```diff
@@ -281,16 +281,25 @@ static void worker(
             else if (v->type() == nAttrs)
               {
                 auto attrs = nlohmann::json::array();
-                StringSet ss;
+                bool recurse = attrPath == "";  // Dont require `recurseForDerivations = true;` for top-level attrset
+
                 for (auto & i : v->attrs->lexicographicOrder()) {
                     std::string name(i->name);
                     if (name.find('.') != std::string::npos || name.find(' ') != std::string::npos) {
                         printError("skipping job with illegal name '%s'", name);
                         continue;
                     }
                     attrs.push_back(name);
+
+                    if (name == "recurseForDerivations") {
+                      auto attrv = v->attrs->get(state.sRecurseForDerivations);
```

Comment:
This will backfire if you have cyclic references: https://github.com/nix-community/nix-eval-jobs/issues/10

Reply:
What I think you're referring to was already the case before.